web/modules/custom/joinup_core/tests/src/Functional/FileUrlFieldTest.php

```diff
@@ -41,36 +44,75 @@ class FileUrlFieldTest extends RdfWebTestBase {
    * Tests upload of a file to an file URL field.
    */
   public function testSingleValuedWidgetLocalFile() {
+    /** @var \Drupal\Core\Entity\ContentEntityStorageInterface $rdf_storage */
+    $rdf_storage = $this->container->get('entity_type.manager')->getStorage('rdf_entity');
+
     $this->drupalLogin($this->rootUser);
 
     $field_name = 'field_file_url';
     $test_file = $this->getTestFile('text');
     $this->assertTrue($test_file instanceof FileInterface);
 
+    $rdf_entity = Rdf::create([
+      'rid' => 'file_url',
+      'label' => 'Foo',
+    ]);
+    $rdf_entity->save();
+
     // Test file for new entities.
-    $this->drupalGet('rdf_entity/add/file_url');
+    $this->drupalGet($rdf_entity->toUrl('edit-form'));
     $this->addFileUrlItem($field_name, 'upload', $test_file->getFileUri());
     $this->drupalPostForm(NULL, ['label[0][value]' => 'Foo'], 'Save');
 
     // Check that the file has been uploaded to the file URL field.
-    $rdf_entity = $this->loadEntityByLabel('rdf_entity', 'Foo');
+    $rdf_entity = Rdf::load($rdf_entity->id());
+    $rdf_entity_file = FileUrlHandler::urlToFile($rdf_entity->{$field_name}->target_id);
+    $initial_uri = $rdf_entity_file->getFileUri();
+    $this->assertFileExists($initial_uri);
+
+    // Ensure the file can be downloaded.
+    $this->drupalGet(file_create_url($initial_uri));
+    $this->assertSession()->statusCodeEquals(200);
+
+    // Upload the same file again to test if the file is saved with in new
+    // location while still keeping the same file basename.
+    $this->drupalPostForm($rdf_entity->toUrl('edit-form'), [], 'Remove');
+    $this->addFileUrlItem($field_name, 'upload', $test_file->getFileUri());
+    $this->drupalPostForm(NULL, [], 'Save');
+
+    // @todo We should not need cache clearing here. The cache should have been
+    //   wiped out at this point. Fix this regression in ISAICP-3392.
+    // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-3392
+    $rdf_storage->resetCache([$rdf_entity->id()]);
+
+    // Check that the file has been uploaded to the file URL field.
+    $rdf_entity = Rdf::load($rdf_entity->id());
     $rdf_entity_file = FileUrlHandler::urlToFile($rdf_entity->{$field_name}->target_id);
-    $this->assertFileExists($rdf_entity_file->getFileUri());
+    $final_uri = $rdf_entity_file->getFileUri();
+    $this->assertFileExists($final_uri);
 
     // Ensure the file can be downloaded.
-    $this->drupalGet(file_create_url($rdf_entity_file->getFileUri()));
+    $this->drupalGet(file_create_url($final_uri));
     $this->assertSession()->statusCodeEquals(200);
 
+    // Check that the same file is uploaded to different locations.
+    $this->assertNotEquals($initial_uri, $final_uri);
```

Comment:
Is this important to be tested? This seems like an implementation detail, because the persisting of the original filenames can also be solved in a different way.

On the other hand, this also verifies that the second file hasn't overwritten the first file, so it is probably useful after all.